Sources/Testing/Running/Runner.swift

```diff
@@ -114,6 +114,50 @@ extension Runner {
     }
   }
 
+  /// Execute the ``CustomExecutionTrait/execute(_:for:testCase:)`` functions
+  /// associated with the test in a plan step.
+  ///
+  /// - Parameters:
+  ///   - step: The step being performed. If this argument is `nil`, `body` is
+  ///     immediately executed.
+  ///   - testCase: The test case, if applicable, for which to execute the
+  ///               custom trait.
+  ///   - body: A function to execute from within the
+  ///           ``CustomExecutionTrait/execute(_:for:testCase:)`` functions of
+  ///           each trait applied to `step.test`.
+  ///
+  /// - Throws: Whatever is thrown by `body` or by any of the
+  ///   `       `CustomExecutionTrait/execute(_:for:testCase:)`` functions.
+  private func _executeTraits(
+    for step: Plan.Step,
+    testCase: Test.Case?,
+    _ body: @escaping @Sendable () async throws -> Void
+  ) async throws {
+    if step.test.traits.isEmpty {
+      return try await body()
+    }
+
+    if case .skip = step.action {
+      return try await body()
```

Comment:
This `if` basically says "don't run `execute(_:test:testCase:)` if the test is to be skipped, which we should document in the documentation for `CustomExecutionTrait`.